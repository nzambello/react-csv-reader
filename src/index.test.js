import React from 'react'
import renderer from 'react-test-renderer'
import CSVReader from './index'

test('Renders basic CSVReader', () => {
  const component = renderer.create(<CSVReader onFileLoaded={() => {}} />)

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Renders CSVReader with all custom props', () => {
  const papaparseOptions = {
    header: true,
    dynamicTyping: true,
    skipEmptyLines: true,
    transformHeader: header => header.toLowerCase().replace(/\W/g, '_'),
  }

  const component = renderer.create(
    <CSVReader
      accept=".csv, text/csv, .tsv, test/tsv"
      cssClass="custom-csv-reader"
      cssInputClass="custom-csv-input"
      fileEncoding="iso-8859-1"
      inputId="react-csv-reader"
      inputStyle={{ color: 'red' }}
      label="Select CSV with secret Death Star statistics"
      onError={e => console.error(e)}
      onFileLoaded={(data, fileName) => console.log(data, fileName)}
      parserOptions={papaparseOptions}
    />,
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
