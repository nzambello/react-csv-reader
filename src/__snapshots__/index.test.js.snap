// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders CSVReader with all custom props 1`] = `
<div
  className="custom-csv-reader"
>
  <label
    htmlFor="react-csv-reader"
  >
    Select CSV with secret Death Star statistics
  </label>
  <input
    accept=".csv, text/csv, .tsv, test/tsv"
    className="custom-csv-input"
    id="react-csv-reader"
    onChange={[Function]}
    style={
      Object {
        "color": "red",
      }
    }
    type="file"
  />
</div>
`;

exports[`Renders basic CSVReader 1`] = `
<div
  className="csv-reader-input"
>
  <input
    accept=".csv, text/csv"
    className="csv-input"
    id="react-csv-reader-input"
    onChange={[Function]}
    style={Object {}}
    type="file"
  />
</div>
`;
